Sources/NIOSSL/SSLCertificate.swift

```diff
@@ -46,6 +46,16 @@ public class NIOSSLCertificate {
         case ipv4(in_addr)
         case ipv6(in6_addr)
     }
+    
+    public lazy var serialNumber: String = {
+        let serialNumber = CNIOBoringSSL_X509_get_serialNumber(self.ref)
+        var bn = BIGNUM()
```

Comment:
This BIGNUM based solution is a bit roundabout, and has cleanup concerns which we aren't addressing. I think we can do a bit better.

BoringSSL stores ASN.1 integers internally as "a big-endian representation of the data, ignoring the sign". The sign information is stored in the type detail. We also know, per X.509's requirements, that the serial number is not more than 20 bytes long, and must be positive.

With those constraints we can say that we'll have an easier time doing the hex-encoding ourselves and avoiding the dance into BoringSSL land. I think we want this:

```swift
        let serial = CNIOBoringSSL_X509_get_serialNumber(self.ref)!
        return Array(UnsafeBufferPointer(start: serial.pointee.data, count: Int(serial.pointee.length)))
```